ConfigurableProduct show prices in select options

arendarenko opened the pull request:
### Description
In Magento2 by default we can't show configurable product prices in select options, as it was done in custom options. There only labels with product name there. My improvement fixes that problem. 

### Manual testing scenarios
1. Add some configurable products to product in backend. Some options should have different price than another.
2. Open product page and click on the select which contains allowed products

**Expected result:**
You can see labels with prices

**Actual result:**
You can see labels without prices, but when you choose some of label, product price is updating

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

arendarenko committed:
ConfigurableProduct module fixes

- Add feature to show price of customized product in select on product page

magento-engcom-team commented:
Hi @alexeya-ven. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

arendarenko commented on app/code/Magento/ConfigurableProduct/view/frontend/web/js/configurable.js:
```diff
@@ -387,14 +394,27 @@ define([
                         options[i].allowedProducts = allowedProducts;
                         element.options[index] = new Option(this._getOptionLabel(options[i]), options[i].id);
 
                         if (typeof options[i].price !== 'undefined') {
```
Looks like this functionality is never used. Also it has a bug with attribute name (uses price**s** instead price)

magento-engcom-team commented:
Hi @VladimirZaets, thank you for the review.
[ENGCOM-2832](https://jira.corp.magento.com/browse/ENGCOM-2832) has been created to process this Pull Request

arendarenko committed:
ConfigurableProduct module fixes

- Fix isShowSign parameter passing that adds "+" or "-" to the option. Previously there was a bug because this parameter should be boolean.

arendarenko committed:
ConfigurableProduct module fixes

- Remove unused variable

arendarenko committed:
ConfigurableProduct module fixes

- Fix code style errors

ihor-sviziev committed:
magento/magento2#17695 ConfigurableProduct show prices in select options 

Fix js static test failure

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-2832](https://jira.corp.magento.com/browse/ENGCOM-2832) has been created to process this Pull Request

alexeya-ven commented:
Hi @VladimirZaets 
I've tried to reproduce this error and found that price is not appearing because by default Magento shows minimal price of product (which is $39), so near this option value it didn't shows any extra-price. In my case it looks like this: 
![screenshot from 2018-08-31 14-35-18](https://user-images.githubusercontent.com/42036143/44910250-2c1b8480-ad2b-11e8-8f18-3f11e95d573f.png)

magento-engcom-team commented:
Hi @alexeya-ven. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.8. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.
